### ISSUE TITLE
Unsafe instructions iteration

### ISSUE THREAD

**Issue by lkubik** [find-sec-bugs/find-sec-bugs#19]
Hi,

I've tried to integrate your plugin to the EAP findbugs tests in jenkins but then I've got into some troubles. When I run the tests without the find-sec-bugs plugin, it ends with a couple of warnings but when I run it with the plugin then I get 42 analysis errors and a really long stack trace.

I run the tests inside the Jenkins and I use the Findbugs in the version 2.0.1. Find-sec-bugs is in the version 1.2.1. Are there any version compability notes? Are these version compatible?

I also didn't find any settings recommendation. I've just put your plugin inside the Findbugs plugin directory. Does it need any additionally settings?

Here is the stack:
https://drive.google.com/file/d/0ByAl3ieZYVnXRWZsbHB6UWFYaGM/view?usp=sharing

```
Nov 03, 2014 5:20:56 AM edu.umd.cs.findbugs.TextUIBugReporter reportAnalysisError
SEVERE: Oops
edu.umd.cs.findbugs.ba.DataflowAnalysisException: Accessing TOP or BOTTOM frame!
    at edu.umd.cs.findbugs.ba.Frame.getStackValue(Frame.java:235)
    at edu.umd.cs.findbugs.ba.npe.IsNullValueFrameModelingVisitor.visitPUTFIELD(IsNullValueFrameModelingVisitor.java:323)
    at org.apache.bcel.generic.PUTFIELD.accept(PUTFIELD.java:79)
    at edu.umd.cs.findbugs.ba.AbstractFrameModelingVisitor.analyzeInstruction(AbstractFrameModelingVisitor.java:84)
    at edu.umd.cs.findbugs.ba.npe.IsNullValueFrameModelingVisitor.analyzeInstruction(IsNullValueFrameModelingVisitor.java:104)
    at edu.umd.cs.findbugs.ba.npe.IsNullValueAnalysis.transferInstruction(IsNullValueAnalysis.java:326)
    at edu.umd.cs.findbugs.ba.npe.IsNullValueAnalysis.transferInstruction(IsNullValueAnalysis.java:71)
[...]
```

Do you have any suggestions how to solve this?

Regards,
Lukas Kubik

**Comment by lkubik** [comment]
Hi @h3xstream ,

first of all I would like to thank you for the bug fix. Also big thanks to @amaembo for a great support.

I tested the fixed version and it works OK now. All of the "oops" exceptions are gone. However there is a one new exception. It appears when I run the analysis of the WildFly code:

```
  Exception analyzing org.apache.xalan.xslt.Process using detector com.h3xstream.findsecbugs.xxe.SaxParserXxeDetector
    java.lang.RuntimeException: edu.umd.cs.findbugs.ba.MethodUnprofitableException: Appears unprofitable to analyze org.apache.xalan.xslt.Process.main(String[])
      At com.h3xstream.findsecbugs.xxe.SaxParserXxeDetector.sawOpcode(SaxParserXxeDetector.java:106)
      At edu.umd.cs.findbugs.visitclass.DismantleBytecode.visit(DismantleBytecode.java:808)
      At edu.umd.cs.findbugs.visitclass.BetterVisitor.visitCode(BetterVisitor.java:217)
      At edu.umd.cs.findbugs.visitclass.PreorderVisitor.visitCode(PreorderVisitor.java:223)
      At edu.umd.cs.findbugs.bcel.OpcodeStackDetector.visitCode(OpcodeStackDetector.java:55)
      At org.apache.bcel.classfile.Code.accept(Code.java:133)
      At edu.umd.cs.findbugs.visitclass.PreorderVisitor.doVisitMethod(PreorderVisitor.java:293)
      At edu.umd.cs.findbugs.visitclass.PreorderVisitor.visitJavaClass(PreorderVisitor.java:373)
      At org.apache.bcel.classfile.JavaClass.accept(JavaClass.java:214)
      At edu.umd.cs.findbugs.BytecodeScanningDetector.visitClassContext(BytecodeScanningDetector.java:37)
      At edu.umd.cs.findbugs.DetectorToDetector2Adapter.visitClass(DetectorToDetector2Adapter.java:74)
      At edu.umd.cs.findbugs.FindBugs2.analyzeApplication(FindBugs2.java:1193)
      At edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:279)
      At edu.umd.cs.findbugs.FindBugs.runMain(FindBugs.java:391)
      At edu.umd.cs.findbugs.FindBugs2.main(FindBugs2.java:1300)
      At sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
      At sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
      At sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
      At java.lang.reflect.Method.invoke(Method.java:606)
      At edu.umd.cs.findbugs.FindBugsMain.invoke(FindBugsMain.java:61)
      At edu.umd.cs.findbugs.workflow.FB.main(FB.java:50)
```
What are the reasons to mark a method as an unprofitable to analyze?
Source of the method is here:
http://svn.apache.org/viewvc/xalan/java/trunk/src/org/apache/xalan/xslt/Process.java?view=markup
If you run this shell script, it will reproduce that exception:
https://drive.google.com/file/d/0ByAl3ieZYVnXZlJCV0ZOSmZ1SU0/view?usp=sharing

**Comment by amaembo** [comment]
Hello! @lkubik, this Process.main method is too big. By default methodGen/CFG analysis is skipped for methods having bytecode longer than 6000 bytes. Otherwise analysis might become too slow and/or memory consuming. There's probably an option to switch this off, but I'm on vacation now so I cannot look in detail.

@h3xstream, it's probably better to silently ignore MethodUnprofitableException [here](https://github.com/h3xstream/find-sec-bugs/blob/master/plugin/src/main/java/com/h3xstream/findsecbugs/xxe/SaxParserXxeDetector.java#L106) instead of rethrowing.

**Comment by h3xstream** [comment]
The analysis can't be done due to this missing method analysis.
I could log an error with no stacktrace and return immediately.

The change would be made at those places ..
(https://github.com/h3xstream/find-sec-bugs/search?utf8=%E2%9C%93&q=CFGBuilderException+)

**Comment by lkubik** [comment]
@amaembo Thanks for the explanation. I used the command line option -effort:max and the exception is no longer thrown. So I guess the limit is off now. We analyze it on a relatively strong dedicated machines so I think that it shouldn't be a problem.

@h3xstream I opened new issue here: https://github.com/h3xstream/find-sec-bugs/issues/29
but it doesn't have a high priority because I use the FindBugs with max effort now...
